src/DateRangePicker.jsx

```diff
@@ -90,11 +90,17 @@ const DateRangePicker = React.createClass({
   componentWillReceiveProps(nextProps) {
     var nextDateStates = this.getDateStates(nextProps);
     var nextEnabledRange = this.getEnabledRange(nextProps);
-
-    this.setState({
+    var updatedState = {
       dateStates: this.state.dateStates && Immutable.is(this.state.dateStates, nextDateStates) ? this.state.dateStates : nextDateStates,
       enabledRange: this.state.enabledRange && this.state.enabledRange.isSame(nextEnabledRange) ? this.state.enabledRange : nextEnabledRange,
-    });
+    };
+
+    if (nextProps.value ){
```

Comment:
@AlanFoster I am half way towards a solution, as well as being able to set range externally but will not get it complete for a couple of weeks , are you happy to just leave the PR open till then? Or close and I can re-submit when ready if that would be preferable. Thanks